src/main/java/com/lalabrand/ecommerce/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,126 @@
+import com.lalabrand.ecommerce.order.shipping.shipping_info.ShippingInfoRequest;
+import com.lalabrand.ecommerce.user.User;
+import com.lalabrand.ecommerce.user.UserRepository;
+import com.lalabrand.ecommerce.user.cart.CartDTO;
+import com.lalabrand.ecommerce.user.cart.CartService;
+import com.lalabrand.ecommerce.user.cart.cart_item.CartItem;
+import com.lalabrand.ecommerce.utils.CommonResponse;
+import com.lalabrand.ecommerce.utils.CommonUtils;
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+
+    private final OrderRepository orderRepository;
+    private final UserRepository userRepository;
+    private final CartService cartService;
+    private final OrderItemsServiceImpl orderItemsService;
+    private final ShippingInfoRepository shippingInfoRepository;
+
+    @Autowired
+    public OrderServiceImpl(OrderRepository orderRepository, UserRepository userRepository, CartService cartService, OrderItemsServiceImpl orderItemsService, ShippingInfoRepository shippingInfoRepository) {
+        this.orderRepository = orderRepository;
+        this.userRepository = userRepository;
+        this.cartService = cartService;
+        this.orderItemsService = orderItemsService;
+        this.shippingInfoRepository = shippingInfoRepository;
+    }
+
+
+    @Transactional
+    @Override
+    public CommonResponse placeOrder(String userId, ShippingInfoRequest shippingInfoRequest) {
+        CartDTO cartDto = cartService.findCartByUserId(userId).orElseThrow(
+                () -> new EntityNotFoundException("Cart hasn't been found for user ( it's empty )")
+        );
+
+        ShippingInfo shippingInfo = ShippingInfo.builder()
+                .address1(shippingInfoRequest.getAddress1())
+                .address2(shippingInfoRequest.getAddress2())
+                .city(shippingInfoRequest.getCity())
+                .country(shippingInfoRequest.getCountry())
+                .zip(shippingInfoRequest.getZip())
+                .phone(shippingInfoRequest.getPhone())
+                .shippingOptionId(shippingInfoRequest.getShippingOptionId())
+                .build();
+
+        ShippingInfo savedShippingInfo = shippingInfoRepository.save(shippingInfo);
+
+        Set<OrderedItem> orderedItems = new HashSet<>();
+        Set<CartItem> cartItems = cartDto.toEntity(userRepository.getReferenceById(userId)).getCartItems();
+
+        Order order = Order.builder()
+                .user(userRepository.getReferenceById(userId))
+                .orderNumber(CommonUtils.getNext())
+                .status(Status.PENDING)
+                .currency(Currency.UAH) // need
```

Comment:
Why we dont get a currency for shipping from front end? I thing user must choose a currecy